ui/admin/app/components/form/auth-method/oidc/index.js

```diff
@@ -27,4 +30,48 @@ export default class FormAuthMethodOidcComponent extends Component {
    * @type {string}
    */
   newSigningAlgorithm = options.oidc.signing_algorithms[0];
+
+  /**
+   * @type {string}
+   */
+  prompts = options.oidc.prompts;
+
+  @tracked selectedPrompts = this.parsePromptsArray();
+
+  /**
+   * @returns {string}
+   */
+  parsePromptsArray() {
+    return (this.args.model.prompts || []).map((item) => item.value);
+  }
+
+  //actions
+
+  /**
+   * @param {string} option
+   * @param {object} event
+   */
+  @action
+  updatePrompt(option, event) {
+    const currentSelection = option.split(',').map((value) => ({ value }));
+
+    const previousSelection = this.args.model.prompts || [];
+
+    //when the checkbox is unchecked, remove the option form the exisiting list
+    const removeSelection = previousSelection.filter((i) => i.value !== option);
+
+    //add the selected option to the model, only when the checkbox is in checked state
+    if (event.target.id && event.target.checked) {
+      if (previousSelection) {
+        set(this.args.model, 'prompts', [
+          ...previousSelection,
+          ...currentSelection,
+        ]);
+      } else {
+        set(this.args.model, 'prompts', [...currentSelection]);
+      }
+    } else {
+      set(this.args.model, 'prompts', [...removeSelection]);
+    }
+  }
```

Comment:
Couldn't this have been simplified by doing an if check on `checked` and if it's true, adding it to the model and if it's false, removing it from the model?

Do we also need to use ember `set`? I know we've used it previously but is it needed anymore?

Reply:
set seemed like the way to add the value of a property to the model,  What would we use in place of `set`?

Reply:
Can't you just set it directly? i.e. `this.args.model.prompts = `